src/lib.rs

```diff
@@ -21,19 +21,40 @@ pub enum WeightUnit {
 
 fn setup_store<T>(store: &mut Store<T>, weight_unit: WeightUnit) {
     store.set_epoch_deadline(1);
-    store.epoch_deadline_callback(move |context| {
-        if let Some(ticker) = unsafe { TICKER.as_ref() } {
-            let mut backtraces = BACKTRACES.lock().unwrap();
-            let weight = match weight_unit {
-                WeightUnit::Nanoseconds => ticker.duration().as_nanos(),
-                WeightUnit::Fuel => context.fuel_consumed().unwrap_or(0).into(),
-            };
-            let last_weight = *LAST_WEIGHT.lock().unwrap();
-            *LAST_WEIGHT.lock().unwrap() = weight;
-            backtraces.push((WasmBacktrace::capture(&context), weight - last_weight));
+
+    match weight_unit {
+        WeightUnit::Fuel => {
```

Comment:
Moved the match to the top level since the fuel case has a fuel loading step prior to establishing the deadline callback.

I extracted the backtrace push into a separate function, but didn't manage to extract the `epoch_deadline_callback` itself, since it raised a `functions are not safe to pass between threads` error. 

If Rustaceans have other ideas on how to best organize this, I'm all ears